src/modules/account/RadixAccount.ts

```diff
@@ -183,15 +220,97 @@ export default class RadixAccount {
     }
 
     private _onAtomReceived = async (atomObservation: RadixAtomObservation) => {
-        this.processingAtomCounter.next(this.processingAtomCounter.getValue() + 1)
+
+
+        if (!this.processingAtomCounter.closed && !this.processingAtomCounter.isStopped) {
+            this.processingAtomCounter.next(this.processingAtomCounter.getValue() + 1)
+        } else {
+            logger.error(`☢️ processingAtomCounter closed or stopped`)
+        }
 
         atomObservation.processedData = {}
         for (const system of this.accountSystems.values()) {
             await system.processAtomUpdate(atomObservation)
         }
 
-        this.processingAtomCounter.next(this.processingAtomCounter.getValue() - 1)
+        if (!this.processingAtomCounter.closed && !this.processingAtomCounter.isStopped) {
+            this.processingAtomCounter.next(this.processingAtomCounter.getValue() - 1)
+        } else {
+            logger.error(`☢️ processingAtomCounter closed or stopped`)
+        }
+
     }
 
-    
+    public async requestRadsForDevelopmentFromFaucetService(): Promise<RadixTransaction> {
+        return this.requestRadsForDevelopmentFromFaucetServiceWithoutBalanceUpdate()
```

Comment:
I think we can use names that are shorter but descriptive enough?
E.g "requestFromFaucet"

Reply:
I what it to be clear that this is a testing/development thing... how should we do that? In iOS world I would mark so that this method is only available during development by using a special flag to the compiler - so that the code is not even there for non dev builds, bu writing

```swift
#if DEBUG

// This method is ONLY available during development thanks to compiler magic.
func requestTokensFromFaucet() {
}

#endif
```

Is that possible in Node/TS/JS world?

If not I would like to make it clear that it is a dev function in the name, but might try to make it shorter, what about `requestTestTokensFromFaucet` ??

Reply:
OK, `requestTestTokensFromFaucet` is fine I think